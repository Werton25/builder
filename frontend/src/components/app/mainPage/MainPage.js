import React from 'react';
import PropTypes from 'prop-types';
import Requests from "./requests/Requests";
import MethodForm from "./methodForm/MethodForm";
import {
  exportRequests,
  importRequests
} from "../../../actions/requests";
import {connect} from 'react-redux';
import {getCurrentUri} from "../../../selectors/router";
import {getCurrentRequest} from "../../../selectors/requests";

class MainPage extends React.Component {
  render() {
    if (!this.props.currentRequest) {
      return <div/>
    }
    const {currentUri, exportRequests, importRequests} = this.props;
    return (
      <div className='container-fluid'>
        <div className='panel panel-default'>
          <div className='panel-heading'>
            <div className='row'>
              <div className='col-md-4 col-sm-4'>
                <b>Выберите тестовый запрос:</b>
              </div>
              <div className='col-md-4 col-sm-4'>
                <b>Добавление запроса в список:</b>
              </div>
            </div>
            <div className='row'>
              <Requests/>
              <div className='col-md-2 col-md-offset-2 col-sm-3 col-sm-offset-1'>
                <div className='btn-group pull-right visible-md-inline-block'>
                  <button className='btn btn-default'>
                    <i className='glyphicon glyphicon-download' role='button' title='Скачать наборы'
                       onClick={() => exportRequests(currentUri)}/>
                  </button>
                  <label className='btn btn-default' title='Загрузить наборы'>
                    <i className='glyphicon glyphicon-upload'/>
                    <input className='hidden' type='file' onChange={(e) => importRequests(e.target.files[0])}/>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div className='panel-body'>
            <MethodForm/>
          </div>
        </div>
      </div>
    );
  }
}

MainPage.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object,
  exportRequests: PropTypes.func.isRequired,
  importRequests: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state)
});

const mapDispatchToProps = {
  exportRequests,
  importRequests
};

export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
