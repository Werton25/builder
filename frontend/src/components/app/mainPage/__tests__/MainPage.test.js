import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import MainPage from "../MainPage";

jest.mock('../../buttons/ImportButton');
jest.mock('../requests/Requests');
jest.mock('../methodForm/MethodForm');
jest.mock('../../../../selectors/router', () => ({getCurrentUri: () => '/service/method'}));
jest.mock('../../../../selectors/requests', () => ({
  getCurrentRequest: () => ({id: '1', name: 'test', value: '{}'}),
  getCurrentRequests: () => ({})
}));
jest.mock('connected-react-router');
const store = configureStore()({});

describe('MainPage tests', () => {
  it('MainPage renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MainPage/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

