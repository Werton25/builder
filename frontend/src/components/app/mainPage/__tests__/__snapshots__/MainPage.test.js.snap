// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage tests MainPage renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-4"
        >
          <b>
            Выберите тестовый запрос:
          </b>
        </div>
        <div
          className="col-md-4 col-sm-4"
        >
          <b>
            Добавление запроса в список:
          </b>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-2 col-md-offset-2 col-sm-3 col-sm-offset-1"
        >
          <div
            className="btn-group pull-right visible-md-inline-block"
          >
            <button
              className="btn btn-default"
            >
              <i
                className="glyphicon glyphicon-download"
                onClick={[Function]}
                role="button"
                title="Скачать наборы"
              />
            </button>
            <label
              className="btn btn-default"
              title="Загрузить наборы"
            >
              <i
                className="glyphicon glyphicon-upload"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-body"
    />
  </div>
</div>
`;
